units/commit_queue.go

```diff
@@ -66,6 +67,40 @@ func NewCommitQueueJob(env evergreen.Environment, queueID string, id string) amb
 	return job
 }
 
+func (j *commitQueueJob) TryUnstick(cq *commitqueue.CommitQueue) {
+	//unstuck the queue if the patch is done.
+	nextItem := cq.Next()
+	if nextItem == nil {
+		return
+	}
+	patchIdString := nextItem.Issue
+	if !patch.IsValidId(patchIdString) {
+		j.dequeue(cq, nextItem)
+		j.logError(nil, "The Patch id '%s' is not an object id. It was removed from the queue.", nextItem)
+		return
+	}
+	patchDoc, err := patch.FindOne(patch.ById(patch.NewId(nextItem.Issue)).WithFields(patch.FinishTimeKey))
+	if err != nil {
+		j.AddError(errors.Wrapf(err, "error finding the patch for %s", j.QueueID))
+		return
+	}
+	if patchDoc == nil {
+		j.dequeue(cq, nextItem)
+		j.logError(err, "The patch on top of the queue is nil. It was removed from the queue.", nextItem)
+		return
+	}
+
+	//patchisdone
+	patchFinishTime := patchDoc.FinishTime
```

Comment:
Nit: no reason to have this as a separate variable (same with status--also is that the status? Are we sure it's MergeTestSucceeded? We should check the status probably to determine if it may be failed)